test/e2e/performance_test.go

```diff
@@ -0,0 +1,371 @@
+}
+
+func httpRequest(times int, workloadAmount int, data *TestData, b *testing.B) {
+	nginxPodIP, _ := setupPerformanceTestPods(data, b)
+	// enable pods connectivity policy first
+	if err := setupPerformanceTestPodsConnection(data); err != nil {
+		b.Fatalf("Error when adding network policy to set up connection between performance test pods")
+	}
+	b.Log("Populating performance test workloadAmount")
+	if err := populateWorkload(generateWorkload(workloadAmount), data); err != nil {
+		b.Fatalf("Error when populating workload: %v", err)
+	}
+
+	select { // wait workloads to be realized
+	case <-time.After(*realizeTimeout):
+		b.Fatal("Waiting network policies realization timeout after", realizeTimeout.String())
+	case err := <-checkRealize(workloadAmount, data):
+		if err != nil {
+			b.Fatalf("Checking network policies realization failed: %v", err)
+		}
+	}
+
+	serverURL := &url.URL{Scheme: "http", Host: nginxPodIP, Path: "/"}
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		b.Logf("Running http request bench %d/%d", i+1, b.N)
+		cmd := []string{"ab", "-n", fmt.Sprint(times), "-c", fmt.Sprint(*httpConcurrency), serverURL.String()}
+		if stdout, stderr, err := data.runCommandFromPod(testNamespace, performanceABPodName, abName, cmd); err != nil {
+			b.Errorf("Error when running http request %dx: %v, stdout: %s, stderr: %s\n", times, err, stdout, stderr)
+		}
+	}
+}
+
+func hackBenchTimer(duration time.Duration, b *testing.B) {
+	elem := reflect.ValueOf(b).Elem()
+	durationField := elem.FieldByName("duration")
+	durationField = reflect.NewAt(durationField.Type(), unsafe.Pointer(durationField.UnsafeAddr())).Elem()
+	durationField.Set(reflect.ValueOf(duration))
+}
+
+func dumpFlows(data *TestData) (string, error) {
+	antreaPodName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{"ovs-ofctl", "dump-flows", "br-int"}
+	if stdout, _, err := data.runCommandFromPod(AntreaNamespace, antreaPodName, "antrea-agent", cmd); err != nil {
+		return "", err
+	} else {
+		return stdout, nil
+	}
+}
+
+func checkRealize(workloads int, data *TestData) <-chan error {
+	errCh := make(chan error, 1)
+	failedHit := 0
+	maximumTolerateTimes := 3
+	go func() {
+		for {
+			if flowNums, err := countFlows(data); err != nil {
+				failedHit += 1
+				if failedHit == maximumTolerateTimes {
+					errCh <- fmt.Errorf("dumping flow keeps failed %d times", maximumTolerateTimes)
+					return
+				}
+			} else if flowNums > workloads {
+				errCh <- nil
+				return
+			}
+			failedHit = 0 // reset the counter
+		}
+	}()
+	return errCh
+}
+
+func networkPolicyRealize(workloads int, data *TestData, b *testing.B) {
+	setupPerformanceTestPods(data, b)
+	delay := time.Duration(0)
+	for i := 0; i < b.N; i++ {
+		go func() {
+			if err := populateWorkload(generateWorkload(workloads), data); err != nil {
+				b.Fatalf("Error when populating workload: %v", err)
+			}
+		}()
+
+		start := time.Now()
+		select {
```

Comment:
same as above

Please use one of the utility functions in https://godoc.org/k8s.io/apimachinery/pkg/util/wait for this if applicable.

Reply:
Done.